src/PlaywrightSharp.Tests/Evaluation/PageEvaluateTests.cs

```diff
@@ -0,0 +1,518 @@
+        }
+
+        ///<playwright-file>evaluation.spec.js</playwright-file>
+        ///<playwright-describe>Page.evaluate</playwright-describe>
+        ///<playwright-it>should transfer arrays as arrays, not objects</playwright-it>
+        [Fact]
+        public async Task ShouldTransferArraysAsArraysNotObjects()
+        {
+            var result = await Page.EvaluateAsync<bool>("a => Array.isArray(a)", new[] { 1, 2, 3 });
+            Assert.True(result);
+        }
+
+        ///<playwright-file>evaluation.spec.js</playwright-file>
+        ///<playwright-describe>Page.evaluate</playwright-describe>
+        ///<playwright-it>should modify global environment</playwright-it>
+        [Fact]
+        public async Task ShouldModifyGlobalEnvironment()
+        {
+            await Page.EvaluateAsync("() => window.globalVar = 123");
+            Assert.Equal(123, await Page.EvaluateAsync<int>("globalVar"));
+        }
+
+        ///<playwright-file>evaluation.spec.js</playwright-file>
+        ///<playwright-describe>Page.evaluate</playwright-describe>
+        ///<playwright-it>should evaluate in the page context</playwright-it>
+        [Fact]
+        public async Task ShouldEvaluateInThePageContext()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/global-var.html");
+            Assert.Equal(123, await Page.EvaluateAsync<int>("globalVar"));
+        }
+
+        ///<playwright-file>evaluation.spec.js</playwright-file>
+        ///<playwright-describe>Page.evaluate</playwright-describe>
+        ///<playwright-it>should return undefined for objects with symbols</playwright-it>
+        [Fact]
+        public async Task ShouldReturnUndefinedForObjectsWithSymbols()
+        {
+            Assert.Null(await Page.EvaluateAsync<object>("() => [Symbol('foo4')]"));
+            Assert.Equal(JsonDocument.Parse("{}"), await Page.EvaluateAsync<JsonDocument>(@"() => {
+                var a = { };
+                a[Symbol('foo4')] = 42;
+                return a;
+            }"));
+            Assert.Null(await Page.EvaluateAsync<object>(@"() => {
+                return { foo: [{ a: Symbol('foo4') }] };
+            }"));
+        }
+
+        ///<playwright-file>evaluation.spec.js</playwright-file>
+        ///<playwright-describe>Page.evaluate</playwright-describe>
+        ///<playwright-it>should work with function shorthands</playwright-it>
+        [Fact(Skip = "Not relevant for C#, js specific")]
+        public Task ShouldWorkWithFunctionShorthands()
+        {
+            return Task.CompletedTask;
+        }
+
+        ///<playwright-file>evaluation.spec.js</playwright-file>
+        ///<playwright-describe>Page.evaluate</playwright-describe>
+        ///<playwright-it>should work with unicode chars</playwright-it>
+        [Fact]
+        public async Task ShouldWorkWithUnicodeChars()
+        {
+            var result = await Page.EvaluateAsync<int>("a => a['Σ╕¡µצחσ¡קτ¼ª'], { 'Σ╕¡µצחσ¡קτ¼ª': 42}");

```

Comment:
Something when wrong here. Check this out https://github.com/kblok/puppeteer-sharp/blob/7fca5eeeaa9e138af09e24bd98d11d128cac260b/lib/PuppeteerSharp.Tests/PageTests/EvaluateTests.cs#L80